Tools/TestWebKitAPI/Tests/WebKitCocoa/PermissionsAPI.mm

```diff
@@ -102,4 +124,44 @@ static void urlEncodeIfNeeded(uint8_t byte, StringBuilder& buffer)
     EXPECT_FALSE(didReceiveQueryPermission);
 }
 
+TEST(PermissionsAPI, PermissionMayHaveChanged)
+{
+    [TestProtocol registerWithScheme:@"https"];
+    auto webView = adoptNS([[TestWKWebView alloc] initWithFrame:CGRectMake(0, 0, 320, 500)]);
+
+    wasPermissionStatusChangedToGranted = false;
+    wasPermissionStatusChangedToPrompt = false;
+
+    [webView performAfterReceivingMessage:@"granted" action:^{
+        wasPermissionStatusChangedToGranted = true;
+    }];
+
+    [webView performAfterReceivingMessage:@"prompt" action:^{
+        wasPermissionStatusChangedToPrompt = true;
+    }];
```

Comment:
I would recommend using EXPECT to check if state (message) is expected instead of waiting until these values are true. Timeout is less preferred than failure cause it's harder to debug and it delays the run of other tests. (With current impl, the test can time out if the page sends back "FAIL" message, or if state is unexpected, or if page never sends back message).